mgsr/app.js

```diff
@@ -0,0 +1,398 @@
+      "dishnet34"
+    ],
+    [
+      "920764241886195764",
+      "E-tan"
+    ],
+    [
+      "788325569707245588",
+      "Stink Man"
+    ],
+    [
+      "276134695870267392",
+      "Lheticus"
+    ],
+    [
+      "618908576492027914",
+      "Shmumbz"
+    ],
+    [
+      "600344944070230026",
+      "Ursi"
+    ],
+    [
+      "578332642164867073",
+      "carterferris07"
+    ],
+    [
+      "710503007074386062",
+      "🎄Apple🎄"
+    ],
+    [
+      "525560136811675650",
+      "ChunkyChango"
+    ],
+    [
+      "904871547704070144",
+      "Orangebird"
+    ],
+    [
+      "480296400408805388",
+      "Mitch"
+    ]
+  ];
+
+// Sorts the user IDs by a custom function looking at the names
+userIds.sort(function(x, y) {
+  n1 = normalizeNameForComparison(x[1]);
+  n2 = normalizeNameForComparison(y[1]);
+
+  if (n1 > n2) {
+    return 1;
+  }
+  if (n1 < n2) {
+    return -1;
+  }
+  return 0;
+});
+userIds.push(["tag this person directly in Discord", "(Unlisted User)"]);
+
+const list = document.querySelectorAll("[name='playerlist']");
+
+userIds.forEach((item) => {
+  let option = document.createElement("option");
+  option.value = item[1];
+  option.textContent = item[1];
+  option.setAttribute("data-userid", item[0]);
+  let option2 = option.cloneNode(true);
+  let option3 = option.cloneNode(true);
+  let option4 = option.cloneNode(true);
+  list[0].appendChild(option);
+  list[1].appendChild(option2);
+  list[2].appendChild(option3);
+  list[3].appendChild(option4);
+});
+
+function updateState() {
+  // Update global object with selected values
+  // let firstPlace = document.getElementById("placement1");
+  let places = [
+    document.getElementById("placement1"),
+    document.getElementById("placement2"),
+    document.getElementById("placement3"),
+    document.getElementById("placement4"),
+  ];
+  let userIDs = [
+    places[0].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[1].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[2].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+    places[3].nextElementSibling.selectedOptions[0].getAttribute("data-userid"),
+  ];
+  let userNames = [
+    places[0].nextElementSibling.selectedOptions[0].value,
+    places[1].nextElementSibling.selectedOptions[0].value,
+    places[2].nextElementSibling.selectedOptions[0].value,
+    places[3].nextElementSibling.selectedOptions[0].value,
+  ];
+  Object.assign(gameState, {
+    1: {
+      place: places[0].value,
+      userID: userIDs[0],
+      userName: userNames[0],
+    },
+    2: {
+      place: places[1].value,
+      userID: userIDs[1],
+      userName: userNames[1],
+    },
+    3: {
+      place: places[2].value,
+      userID: userIDs[2],
+      userName: userNames[2],
+    },
+    4: {
+      place: places[3].value,
+      userID: userIDs[3],
+      userName: userNames[3],
+    },
+  });
+
+  renderCommands(gameState);
```

Comment:
Removes on click on users part if the commands just auto-regenerate each time the state changes